lib/puppet/provider/sensor_download/sensor_download.rb

```diff
@@ -18,7 +18,7 @@ def destroy
 
   def exists?
     falcon_version = Facter.value('falcon_version')
-    installed = falcon_version != :purged
+    installed = [:absent, :purged, :undef, nil].include?(falcon_version) ? false : true
```

Comment:
`nil` option disappeared here

but was not added to the list in `facter`.

Reply:
I am not sure if it is real possibility, but it seems that `:ensure` may be missing in the indirection hash. ? WDYT?

Reply:
Hey are you referring to this line `pkg_ensure = Puppet::Resource.indirection.find("package/#{pkg_name}").to_hash[:ensure]` ?

It's not documented, but based on my testing the symbols you can check is determined by the resource so `:ensure` should always be valid since we are checking `package` even if the package doesn't exist. For example a package with the name `fjsdlkfjsflk` would returned `ensure => 'purged'`